packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -123,19 +184,81 @@ class DatePickerInput extends React.Component {
     }
   }
 
-  handleOnChange(event) {
-    if (!DateUtil.validDateInput(event.target.value)) {
-      return;
+  handleOnKeyDown(event) {
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event);
     }
+  }
+
+  handleDayChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.DAY, inputValue);
+  }
 
-    if (this.props.onChange) {
-      this.props.onChange(event);
+  handleMonthChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.MONTH, inputValue);
+  }
+
+  handleYearChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.YEAR, inputValue);
+  }
+
+  handleOnFocus(event) {
+    this.setState({ isFocused: true });
+    if (this.props.onFocus) {
+      this.props.onFocus(event);
     }
   }
 
-  handleOnKeyDown(event) {
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(event);
+  handleOnBlur(event) {
+    this.setState({ isFocused: false });
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
+
+  handleDateChange(event, type, value) {
+    if (type === DateUtil.inputType.DAY) {
+      this.setState({
+        day: value,
+      }, this.moveFocusOnChange(value, type, this.variant));
+    } else if (type === DateUtil.inputType.MONTH) {
+      this.setState({
+        month: value,
+      }, this.moveFocusOnChange(value, type, this.variant));
+    } else if (type === DateUtil.inputType.YEAR) {
+      this.setState({
+        year: value,
+      }, this.moveFocusOnChange(value, type, this.variant));
+    }
+
+    const day = type === DateUtil.inputType.DAY ? value : this.state.day;
```

Comment:
Please add a brief code comment for this blob of code (Lines 237 to 261).

Reply:
Added! [89c3693](https://github.com/cerner/terra-framework/pull/1004/commits/89c3693f31162adfc53f90f681cdfda21588b6b0)